tests/unit/tls.tcl

```diff
@@ -114,5 +114,30 @@ start_server {tags {"tls"}} {
                 }
             }
         }
+
+        test {TLS: switch between tcp and tls ports} {
```

Comment:
Consider increasing coverage:
- Listen failure (can try to listen on a port we know the TCL suite is listening on, etc.), I think this would even catch an issue with the current implementation.
- Add bind change tests.

Reply:
add two more tests `CONFIG SET port number` and `CONFIG SET port number`